src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -109,6 +118,12 @@ public void deleteCertificate(Certificate certificate) {
         runtimeConfiguration.removeCertificateV1(certificate.getCertificateId());
     }
 
+    private boolean isTrusted(Certificate certificate) {
```

Comment:
Ideally we don't have registries trigger use cases, they should just deal with querying data from the store and returning domain objects and saving data. (See previous comment). When we do this we are mixing business logic inside of infra